packages/forms/src/UIForm/fields/ResourcePicker/ResourcePicker.component.js

```diff
@@ -40,18 +42,30 @@ class ResourcePicker extends Component {
 
 	onFilter(event) {
 		this.setState({ isLoading: true }, () => {
-			this.props
-				.onTrigger(event, {
-					trigger: {
-						parameters: this.state.filters,
-					},
-					schema: this.props.schema,
-				})
+			this.onTrigger(event, FILTER, { filters: this.state.filters })
 				.then(data => this.setState(data))
 				.finally(() => this.setState({ isLoading: false }));
 		});
 	}
 
+	onTrigger(event, eventName, payload) {
+		const { schema, properties, errors } = this.props;
+		const trigger = schema.triggers && schema.triggers.find(trig => trig.onEvent === eventName);
+		const formData = mutateValue(properties, schema, payload.value);
```

Comment:
You don't need that anymore :)
It was one of our tries, but in fact you switched to a properties updater, so no need to have the up to date properties.